scripts/tf_cnn_benchmarks/benchmark_cnn.py

```diff
@@ -672,6 +672,9 @@
                     'different test runs. This flag is designed for human '
                     'consumption, and does not have any impact within the '
                     'system.')
+flags.DEFINE_boolean('auto_mixed_precision', False,
```

Comment:
Loss scaling is needed when mixed precision is used. Currently loss scaling is only used when `--use_fp16` is specified, so please change that logic so its also used when `--auto_mixed_precision` is used.

Reply:
As per documentation, dynamic loss scaling is also automatically applied when that option is enabled. From the page :  https://www.tensorflow.org/api_docs/python/tf/train/experimental/enable_mixed_precision_graph_rewrite


> Calling enable_mixed_precision_graph_rewrite(opt) enables the graph rewrite operation before computing gradients. The function additionally returns an Optimizer(opt) wrapped with a LossScaleOptimizer. This prevents underflow in the float16 tensors during the backward pass

Reply:
Dynamic loss scaling is only applied when the function `tf.train.experimental.enable_mixed_precision_graph_rewrite` is called. In your case, you are instead directly enabling the graph rewrite by modifying `config.graph_options.rewrite_options.auto_mixed_precision`, so loss scaling does not automatically occur.

Reply:
You are right. 

I had incorrectly assumed that setting that flag would also enable dynamic loss scaling. I have pushed out a commit with the requested change, please review.

thanks